master/internal/api_runs.go

```diff
@@ -43,3 +56,178 @@ func (a *apiServer) RunPrepareForReporting(
+		if err != nil {
+			return nil, err
+		}
+
+		query = query.Where("e.project_id = ?", req.ProjectId)
+	}
+	if query, err = experiment.AuthZProvider.Get().
+		FilterExperimentsQuery(ctx, *curUser, proj, query,
+			[]rbacv1.PermissionType{rbacv1.PermissionType_PERMISSION_TYPE_VIEW_EXPERIMENT_METADATA},
+		); err != nil {
+		return nil, err
+	}
+
+	if req.Filter != nil {
+		var efr experimentFilterRoot
+		err := json.Unmarshal([]byte(*req.Filter), &efr)
+		if err != nil {
+			return nil, err
+		}
+		query = query.WhereGroup(" AND ", func(q *bun.SelectQuery) *bun.SelectQuery {
+			_, err = efr.toSQL(q)
+			return q
+		}).WhereGroup(" AND ", func(q *bun.SelectQuery) *bun.SelectQuery {
+			if !efr.ShowArchived {
+				return q.Where(`e.archived = false`)
+			}
+			return q
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if req.Sort != nil {
+		err = sortRuns(req.Sort, query)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		query.OrderExpr("id ASC")
+	}
+
+	pagination, err := runPagedBunExperimentsQuery(ctx, query, int(req.Offset), int(req.Limit))
+	resp.Pagination = pagination
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+func getRunsColumns(q *bun.SelectQuery) *bun.SelectQuery {
+	return q.
+		Column("r.id").
+		ColumnExpr("proto_time(r.start_time) AS start_time").
+		ColumnExpr("proto_time(r.end_time) AS end_time").
+		ColumnExpr(bunutils.ProtoStateDBCaseString(trialv1.State_value, "r.state", "state",
+			"STATE_")).
+		Column("r.tags").
+		Column("r.checkpoint_size").
+		Column("r.checkpoint_count").
+		Column("r.external_run_id").
+		ColumnExpr(
+			"(SELECT COUNT(*) FROM runs r WHERE e.id = r.experiment_id) AS num_runs").
+		ColumnExpr("extract(epoch FROM coalesce(r.end_time, now()) - r.start_time)::int AS duration").
+		ColumnExpr("COALESCE(u.display_name, u.username) as display_name").
+		ColumnExpr("r.hparams AS hyperparameters").
+		ColumnExpr("r.summary_metrics AS summary_metrics").
+		ColumnExpr("CASE WHEN e.parent_id IS NULL THEN NULL ELSE " +
+			"json_build_object('value', e.parent_id) END AS forked_from").
+		ColumnExpr("CASE WHEN e.progress IS NULL THEN NULL ELSE " +
+			"json_build_object('value', e.progress) END AS progress").
+		ColumnExpr("e.id AS experiment_id").
+		Column("e.owner_id").
+		ColumnExpr("e.config->>'description' AS description").
+		ColumnExpr("e.config->'resources'->>'resource_pool' AS resource_pool").
+		ColumnExpr("e.config->'searcher'->>'name' AS searcher_type").
+		ColumnExpr("e.config->'searcher'->>'metric' AS searcher_metric").
+		ColumnExpr("e.config->>'name' as experiment_name").
+		ColumnExpr("w.id AS workspace_id").
+		ColumnExpr("w.name AS workspace_name").
+		ColumnExpr("(w.archived OR p.archived) AS parent_archived").
+		Column("e.unmanaged").
+		ColumnExpr("p.name AS project_name").
+		Join("LEFT JOIN experiments AS e ON r.experiment_id=e.id").
+		Join("LEFT JOIN users u ON e.owner_id = u.id").
+		Join("LEFT JOIN projects p ON e.project_id = p.id").
```

Comment:
runs have their own project ID field that we should use instead of going through the experiment